src/test/java/com/synopsys/integration/alert/web/actions/SystemActionsTest.java

```diff
@@ -1,157 +1,88 @@
         Mockito.when(defaultSystemMessageUtility.getSystemMessages()).thenReturn(messages);
+        Mockito.when(defaultSystemMessageUtility.getSystemMessagesBefore(Mockito.any())).thenReturn(messages);
         Mockito.when(defaultSystemMessageUtility.getSystemMessagesAfter(Mockito.any())).thenReturn(messages);
-        Mockito.when(settingsUtility.getKey()).thenReturn(SETTINGS_DESCRIPTOR_KEY);
-        Mockito.when(settingsUtility.doesConfigurationExist()).thenReturn(true);
+        Mockito.when(defaultSystemMessageUtility.findBetween(Mockito.any())).thenReturn(messages);
     }
 
-    @Test
-    public void getSystemMessagesSinceStartup() {
-        SystemActions systemActions = new SystemActions(defaultSystemStatusUtility, defaultSystemMessageUtility, fieldModelProcessor, settingsUtility);
-        systemActions.getSystemMessagesSinceStartup();
-        Mockito.verify(defaultSystemStatusUtility).getStartupTime();
-        Mockito.verify(defaultSystemMessageUtility).getSystemMessagesAfter(Mockito.any());
+    public static List<Pair<String, String>> getStartAndEndTimes() {
+        return List.of(
+            Pair.of(null, null),
+            Pair.of("2018-11-13T00:00:00.000Z", null),
+            Pair.of(null, "2018-11-13T00:00:00.000Z"),
+            Pair.of("2018-11-13T00:00:00.000Z", "2018-11-13T01:00:00.000Z")
+        );
     }
 
-    @Test
-    public void testGetSystemMessagesAfter() throws Exception {
-        SystemActions systemActions = new SystemActions(defaultSystemStatusUtility, defaultSystemMessageUtility, fieldModelProcessor, settingsUtility);
-        systemActions.getSystemMessagesAfter("2018-11-13T00:00:00.000Z");
-        Mockito.verify(defaultSystemMessageUtility).getSystemMessagesAfter(Mockito.any());
-    }
-
-    @Test
-    public void testGetSystemMessagesBefore() throws Exception {
-        SystemActions systemActions = new SystemActions(defaultSystemStatusUtility, defaultSystemMessageUtility, fieldModelProcessor, settingsUtility);
-        systemActions.getSystemMessagesBefore("2018-11-13T00:00:00.000Z");
-        Mockito.verify(defaultSystemMessageUtility).getSystemMessagesBefore(Mockito.any());
-    }
-
-    @Test
-    public void testGetSystemMessagesBetween() throws Exception {
-        SystemActions systemActions = new SystemActions(defaultSystemStatusUtility, defaultSystemMessageUtility, fieldModelProcessor, settingsUtility);
-        systemActions.getSystemMessagesBetween("2018-11-13T00:00:00.000Z", "2018-11-13T01:00:00.000Z");
-        Mockito.verify(defaultSystemMessageUtility).findBetween(Mockito.any());
-    }
-
-    @Test
-    public void testGetSystemMessages() {
-        SystemActions systemActions = new SystemActions(defaultSystemStatusUtility, defaultSystemMessageUtility, fieldModelProcessor, settingsUtility);
-        systemActions.getSystemMessages();
-        Mockito.verify(defaultSystemMessageUtility).getSystemMessages();
+    @ParameterizedTest
```

Comment:
Are we sure we want to use a parameterized test here? It looks like we just want to create a start and end time list and iterate over it in the test. Parameterized tests will just make it harder to debug the test in my opinion.

Reply:
This is what parameterized tests are meant for. I don't think we should change it

Reply:
Parameterized tests are expanded into individual tests where the input is shown as part of the test.

Reply:
Parameterized are meant for stuff like this, but ends up making things harder in my opinion. Why use a parameterized test when you can just pass the list yourself?